### PR TITLE
Autonomous

### PR DESCRIPTION
**Pull request by shinthebinn**
- added auto
- many changes and additions to subsystems and commands to make them work with auto while not affecting teleop
- auto routine is pretty good (still needs tuning though)

### CODE REVIEW

**Review by swcaskey — APPROVED** [review]
Good

File: src/main/cpp/commands/Indexer/SetIndexer.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+// Called when the command is initially scheduled.
+void SetIndexer::Initialize() {
+	if (!mTargetState)
+		mpIndexer->stopIndexer();
+	else if (mTargetState == MotorDirection::kForward)
+		mpIndexer->indexUp();
+	else
+		mpIndexer->indexDown();
```
Comment (caburum) [inline src/main/cpp/commands/Indexer/SetIndexer.cpp]: This should be a switch statement

File: src/main/cpp/commands/Intake/SetRollers.cpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+// Called when the command is initially scheduled.
+void SetRollers::Initialize() {
+	if (!mTargetState)
+		mpIntake->stopRollers();
+	else if (mTargetState == MotorDirection::kForward)
+		mpIntake->startRollers();
+	else
+		mpIntake->startReverseRollers();
```
Comment (caburum) [inline src/main/cpp/commands/Intake/SetRollers.cpp]: This should be a switch statement

File: src/main/cpp/subsystems/Lights.cpp [hunk 3]
```diff
@@ -2,26 +2,18 @@
 
 Lights* Lights::mpLightsInstance = NULL;
 
-Lights::Lights() {
```
Comment (caburum) [inline src/main/cpp/subsystems/Lights.cpp]: Use version from main, don't overwrite it

File: src/main/cpp/subsystems/Shooter.cpp [hunk 4]
```diff
@@ -1,8 +1,8 @@
 #include "subsystems/Shooter.h"
 
 Shooter::Shooter() : mShooterMotor{ShooterConstants::kShooterMotor} {
-	SetName("Shooter");
-	SetSubsystem("Shooter");
+    SetName("Shooter");
+    SetSubsystem("Shooter");
```
Comment (caburum) [inline src/main/cpp/subsystems/Shooter.cpp]: bad

File: src/main/cpp/subsystems/Shooter.cpp [hunk 5]
```diff
@@ -61,7 +61,7 @@ void Shooter::setShooter(double speed, bool useRpm = false) {
 }
 
 void Shooter::stopShooter() {
-	mSpeed = 0.0;
+    mSpeed = 0.0;
```
Comment (caburum) [inline src/main/cpp/subsystems/Shooter.cpp]: style guide

File: src/main/cpp/subsystems/SwerveDrive.cpp [hunk 6]
```diff
@@ -2,7 +2,7 @@
...
+SwerveDrive::SwerveDrive() : mWheelFL{SwerveDriveConstants::kFLDrive, SwerveDriveConstants::kFLTurn, SwerveDriveConstants::kFLEnc, SwerveDriveConstants::kFLOffset, SwerveDriveConstants::kFLMaxSpeed}, mWheelFR{SwerveDriveConstants::kFRDrive, SwerveDriveConstants::kFRTurn, SwerveDriveConstants::kFREnc, SwerveDriveConstants::kFROffset, SwerveDriveConstants::kFRMaxSpeed}, mWheelBR{SwerveDriveConstants::kBRDrive, SwerveDriveConstants::kBRTurn, SwerveDriveConstants::kBREnc, SwerveDriveConstants::kBROffset, SwerveDriveConstants::kBRMaxSpeed}, mWheelBL{SwerveDriveConstants::kBLDrive, SwerveDriveConstants::kBLTurn, SwerveDriveConstants::kBLEnc, SwerveDriveConstants::kBLOffset, SwerveDriveConstants::kBLMaxSpeed} {
```
Comment (caburum) [inline src/main/cpp/subsystems/SwerveDrive.cpp]: Fix merge

File: src/main/cpp/subsystems/SwerveDrive.cpp [hunk 7]
```diff
@@ -63,7 +63,7 @@ void SwerveDrive::controllerSwerve() {
 	} else {
 		offset = 0;
 	}
-	vectorSwerve(fixInput(mpDriverController->GetLeftX()), fixInput(-mpDriverController->GetLeftY()), fixInput(mpDriverController->GetRightX()), offset);
+	vectorSwerve(mpDriverController->GetLeftX(), -mpDriverController->GetLeftY(), mpDriverController->GetRightX(), offset);
```
Comment (swcaskey) [inline src/main/cpp/subsystems/SwerveDrive.cpp]: Add fixInput

Comment (caburum) [inline src/main/cpp/subsystems/SwerveDrive.cpp]: Add fixInput

File: src/main/include/Constants.h [hunk 8]
```diff
@@ -41,7 +47,12 @@ namespace SwerveDriveConstants {
 
 	const double kMaxSpeed = 1.00;
 
-	const double kDeadzone = 0.1;
```
Comment (caburum) [inline src/main/include/Constants.h:44]: Readd

File: src/main/include/RobotContainer.h [hunk 9]
```diff
@@ -3,16 +3,13 @@
 #include <frc/DriverStation.h>
 #include <frc/Joystick.h>
 #include <frc/XboxController.h>
-#include <frc/shuffleboard/Shuffleboard.h>
-#include <frc/shuffleboard/ShuffleboardLayout.h>
-#include <frc/shuffleboard/ShuffleboardTab.h>
```
Comment (caburum) [inline src/main/include/RobotContainer.h]: Readd

File: src/main/include/RobotContainer.h [hunk 10]
```diff
@@ -36,7 +33,6 @@
 #include "subsystems/Intake.h"
 #include "subsystems/Limelight.h"
 #include "subsystems/Shooter.h"
-#include "subsystems/ShooterHood.h"
```
Comment (caburum) [inline src/main/include/RobotContainer.h]: Readd

File: src/main/include/RobotContainer.h [hunk 11]
```diff
@@ -89,13 +85,13 @@ class RobotContainer {
 
 		frc::SendableChooser<frc2::Command*> mChooser;
 
-		AutonomousCommand mAutonomousCommand;
+		// AutonomousCommand mAutonomousCommand;
```
Comment (caburum) [inline src/main/include/RobotContainer.h]: Delete

File: src/main/include/RobotContainer.h [hunk 12]
```diff
@@ -89,13 +85,13 @@ class RobotContainer {
...
 		static RobotContainer* mRobotContainer;
 
 		void ConfigureButtonBindings();
 
 		// Subsystems and commands
-		SwerveDrive mSwerveDrive{&mDriverController};
```
Comment (caburum) [inline src/main/include/RobotContainer.h]: Readd

File: src/main/include/subsystems/Lights.h [hunk 13]
```diff
@@ -35,6 +34,6 @@ class Lights : public frc2::SubsystemBase {
 		static Lights* mpLightsInstance;
 
 #ifdef ARDUINO
-		frc::SerialPort mArduino{9600, frc::SerialPort::kUSB1};
```
Comment (caburum) [inline src/main/include/subsystems/Lights.h]: Readd

File: src/main/include/subsystems/SwerveDrive.h [hunk 14]
```diff
@@ -13,11 +13,14 @@
 
 class SwerveDrive : public frc2::SubsystemBase {
 	public:
-		SwerveDrive(frc::XboxController *pDriverController);
```
Comment (caburum) [inline src/main/include/subsystems/SwerveDrive.h:16]: Readd

File: src/main/include/subsystems/SwerveDrive.h [hunk 15]
```diff
@@ -13,11 +13,14 @@
...
+		SwerveDrive();
+		void setController(frc::XboxController *pDriverController) {
+			mpDriverController = pDriverController;
+		}
 		double fixInput(double s, bool square = true) {
 			if (square) s = std::copysign(s * s, s);
 
-			if (-SwerveDriveConstants::kDeadzone < s && s < SwerveDriveConstants::kDeadzone)
```
Comment (caburum) [inline src/main/include/subsystems/SwerveDrive.h]: Readd

File: vendordeps/Phoenix.json [hunk 16]
```diff
@@ -1,257 +1,257 @@
 {
```
Comment (caburum) [inline vendordeps/Phoenix.json]: Revert your changes

File: vendordeps/WPILibNewCommands.json [hunk 17]
```diff
@@ -1,37 +1,37 @@
 {
```
Comment (caburum) [inline vendordeps/WPILibNewCommands.json]: Revert your changes

File: vendordeps/navx_frc.json [hunk 18]
```diff
@@ -1,35 +1,35 @@
 {
```
Comment (caburum) [inline vendordeps/navx_frc.json]: Revert your changes

**Review by caburum — CHANGES_REQUESTED** [review]

**Review by caburum — APPROVED** [review]

**Pull request merged**